tests/adapters/switches/juniper_test.py

```diff
@@ -263,6 +263,365 @@ def test_get_vlans_where_vlan_interfaces_can_also_be_called_irb(self):
+
+    def test_get_vlan_with_interface(self):
+        self.netconf_mock.should_receive("get_config").with_args(source="running", filter=is_xml("""
+                <filter>
+                  <configuration>
+                    <vlans>
+                      <vlan>
+                        <vlan-id>20</vlan-id>
+                      </vlan>
+                    </vlans>
+                    <interfaces />
+                  </configuration>
+                </filter>
+            """)).and_return(a_configuration("""
+                <vlans>
+                  <vlan>
+                    <name>WITH-IF</name>
+                    <vlan-id>20</vlan-id>
+                    <l3-interface>vlan.20</l3-interface>
+                  </vlan>
+                </vlans>
+                <interfaces>
+              <interface>
+                <name>ge-0/0/1</name>
+                <unit>
+                  <name>0</name>
+                  <family>
+                    <ethernet-switching>
+                    </ethernet-switching>
+                  </family>
+                </unit>
+              </interface>
+                  <interface>
+                    <name>ge-0/0/1</name>
+                  </interface>
+                  <interface>
+                    <name>vlan</name>
+                    <unit>
+                      <name>20</name>
+                      <family>
+                        <inet>
+                          <address>
+                            <name>1.1.1.1/24</name>
+                          </address>
+                          <filter>
+                            <input>
+                              <filter-name>AC-IN</filter-name>
+                            </input>
+                            <output>
+                              <filter-name>AC-OUT</filter-name>
+                            </output>
+                          </filter>
+                        </inet>
+                      </family>
+                    </unit>
+                    <unit>
+                      <name>40</name>
+                    </unit>
+                    <unit>
+                      <name>70</name>
+                      <family>
+                        <inet>
+                          <address>
+                            <name>2.1.1.1/24</name>
+                          </address>
+                          <address>
+                            <name>4.1.1.1/24</name>
+                          </address>
+                          <address>
+                            <name>3.1.1.1/24</name>
+                          </address>
+                        </inet>
+                      </family>
+                    </unit>
+                  </interface>
+                </interfaces>
+            """))
+
+        vlan = self.switch.get_vlan(20)
+
+        assert_that(vlan.number, equal_to(20))
+        assert_that(vlan.name, equal_to(None))
```

Comment:
The name that is return in the Vlan object is taken from the <description> tag sent from Netconf. In this case, no description is given, so there's no name in the Vlan object.